src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -78,82 +88,44 @@ func (a *StorageMinerActorCode_I) NotifyOfSurprisePoStChallenge(rt Runtime) Invo
 	a._expirePreCommittedSectors(rt)
 
 	h, st := a.State(rt)
-	st.ChallengeStatus().Impl().LastChallengeEpoch_ = rt.CurrEpoch()
+	// update challenge start epoch
+	st.ChallengeStatus().Impl().OnNewChallenge(rt.CurrEpoch())
 	UpdateRelease(rt, h, st)
 	return rt.SuccessReturn()
 }
 
-func (a *StorageMinerActorCode_I) _slashDealsFromFaultReport(rt Runtime, sectorNumbers []sector.SectorNumber, action deal.StorageDealSlashAction) {
-
-	h, st := a.State(rt)
-
-	dealIDs := make([]deal.DealID, 0)
-
-	for _, sectorNo := range sectorNumbers {
-
-		utilizationInfo := st._getUtilizationInfo(rt, sectorNo)
-		activeDealIDs := utilizationInfo.DealExpirationAMT().Impl().ActiveDealIDs()
-		dealIDs = append(dealIDs, activeDealIDs...)
-
-	}
-
-	Release(rt, h, st)
-
-	// TODO: Send(StorageMarketActor, ProcessDealSlash)
-}
-
-// construct FaultReport
-// reset NewTerminatedFaults
-func (a *StorageMinerActorCode_I) _submitFaultReport(
-	rt Runtime,
-	newDeclaredFaults sector.CompactSectorSet,
-	newDetectedFaults sector.CompactSectorSet,
-	newTerminatedFaults sector.CompactSectorSet,
-) {
-	faultReport := &sector.FaultReport_I{
-		NewDeclaredFaults_:   newDeclaredFaults,
-		NewDetectedFaults_:   newDetectedFaults,
-		NewTerminatedFaults_: newTerminatedFaults,
-	}
-
-	rt.Abort("TODO") // TODO: Send(SPA, ProcessFaultReport(faultReport))
-	panic(faultReport)
-
-	// only terminatedFault will be slashed
-	if len(newTerminatedFaults) > 0 {
-		a._slashDealsFromFaultReport(rt, newTerminatedFaults.SectorsOn(), deal.SlashTerminatedFaults)
-	}
+// Called by the cron actor at every tick.
+func (a *StorageMinerActorCode_I) OnCronTick(rt Runtime) InvocOutput {
+	a.CheckSurprisePoStSubmissionHappened(rt)
 
-	h, st := a.State(rt)
-	st.SectorTable().Impl().TerminatedFaults_ = sector.CompactSectorSet(make([]byte, 0))
-	UpdateRelease(rt, h, st)
+	return rt.SuccessReturn()
 }
 
-// construct PowerReport from SectorTable
-// need lastPoSt to search for new expired deals in _updateSectorUtilization
-// where DealExpirationAMT takes in a range of Epoch and return a list of values that expire in that range
-func (a *StorageMinerActorCode_I) _submitPowerReport(rt Runtime, lastPoSt block.ChainEpoch) {
-	h, st := a.State(rt)
-	newExpiredDealIDs := st._updateSectorUtilization(rt, lastPoSt)
-	activePower := st._getActivePower(rt)
-	inactivePower := st._getInactivePower(rt)
+// Will return immediately if the miner
+// has successfully submitted a PoSt (and is no longer challenged)
+// If a Post is missed (either due to faults being not declared on time or
+// because the miner run out of time, every sector is reported as failing
+// for the current proving period.
```

Comment:
❗️ This comment is generally misleading as to the method's behaviour and leads me to think it's not doing what we want it to.

An accurate comment might read:
`// If the miner is currently challenged for surprise PoSt, reports every sector as failing for the current proving period.`

There would seem to be a huge precondition that this is only called when the challenge window has passed, not when it's just opened. But that precondition is not met: this is called on every cron tick.

In any case, details about why PoSt is missed are out of scope here.

Reply:
Why do you say misleading? I think more details in comments is generally good. (cc @zixuanzh who is the original author I believe)

Reply:
replaced anyhow